export-tiles/src/main/java/com/esri/arcgisruntime/sample/exporttiles/MainActivity.java

```diff
@@ -0,0 +1,308 @@
+    android.graphics.Point maxScreenPoint = new android.graphics.Point(minScreenPoint.x + (mMapView.getWidth() * 3),
+        minScreenPoint.y + (mMapView.getHeight() * 3));
+    // convert screen points to map points
+    Point minPoint = mMapView.screenToLocation(minScreenPoint);
+    Point maxPoint = mMapView.screenToLocation(maxScreenPoint);
+    // use the points to define and return an envelope
+    return new Envelope(minPoint, maxPoint);
+  }
+
+  /**
+   * Reverts app state to:
+   * - hide progress and preview elements
+   * - bring the main MapView to the front and
+   * - cancel ExportTileCacheJob
+   */
+  private void cancelDownload() {
+    // change downloading flag
+    mDownloading = false;
+    // show export tiles button
+    mExportTilesButton.setVisibility(View.VISIBLE);
+    // make UI elements related to download invisible
+    mProgressLayout.setVisibility(View.GONE);
+    mProgressTextView.setVisibility(View.GONE);
+    mProgressBar.setVisibility(View.GONE);
+    // reset progress bar
+    mProgressBar.setProgress(0);
+    // make map preview invisible
+    mTileCachePreview.getChildAt(0).setVisibility(View.INVISIBLE);
+    mMapView.bringToFront();
+    mExportTilesButton.setText(getResources().getString(R.string.export_tiles_text));
+    if (mExportTileCacheJob != null) {
+      mExportTileCacheJob.cancel();
+    }
+    // show red preview mask
+    mPreviewMask.bringToFront();
+  }
+
+  /**
+   * Using scale defined by the main MapView and the TiledLayer and an extent defined by viewToExtent() as parameters,
+   * downloads a TileCache locally to the device.
+   */
+  private void initiateDownload() {
+
+    // change text of export tiles button to Cancel
+    mExportTilesButton.setText(getResources().getString(R.string.cancel_export_tiles_text));
+    // update progress text view
+    mProgressTextView.setText(getString(R.string.progress_starting));
+    showProgress();
+    //get the parameters by specifying the selected area,
+    //mapview's current scale as the minScale and tiled layer's max scale as maxScale
+    double minScale = mMapView.getMapScale();
+    double maxScale = mTiledLayer.getMaxScale();
+    // minScale must always be larger than maxScale
+    if (minScale <= maxScale) {
+      minScale = maxScale + 1;
+    }
+    //set the state
+    mDownloading = true;
+    //initialize the export task
+    mExportTileCacheTask = new ExportTileCacheTask(mTiledLayer.getUri());
+    final ListenableFuture<ExportTileCacheParameters> parametersFuture = mExportTileCacheTask
+        .createDefaultExportTileCacheParametersAsync(viewToExtent(), minScale, maxScale);
+    parametersFuture.addDoneListener(new Runnable() {
+      @Override public void run() {
+        try {
+          File file = new File(Environment.getExternalStorageDirectory(),
+              getString(R.string.config_data_sdcard_offline_dir));
+          if (!file.exists()) {
+            file.mkdirs();
```

Comment:
This returns a `boolean` and could fail, you should test for existence of file and respond 

```java
private boolean mDirExists = false;

...

mDirExists = file.exists();
if(mDirExists){
  Log.d(TAG, "Directory exists");
}else{
   mDirExists = file.mkdirs();
}
```

Reply:
added